### ISSUE TITLE
Blog Posts Block and Sharing Buttons - Sharing Button Icons require being enabled on both Posts and Pages to function properly.

### ISSUE THREAD

**Issue by scosgro** [Automattic/wp-calypso#52347]
### Steps to reproduce the behavior

1. On a site that has multiple blog posts published, with featured images, create a page with a Blog Posts Block on it. 
2. To replicate the bug fairly visibly, use this configuration, and publish the page:
![image](https://user-images.githubusercontent.com/30731356/116305637-e5d78700-a758-11eb-8268-8297ef7cd0a0.png)
3. Go to Tools - Marketing - Sharing Buttons, and enable sharing buttons on Posts. Make sure your options are set to one with an Icon, either Icon & Text, or Icon Only.
![image](https://user-images.githubusercontent.com/30731356/116306346-cab94700-a759-11eb-92aa-c3c51cd675c7.png)
4. View the page. The sharing buttons will appear, but the icons will be broken:
![image](https://user-images.githubusercontent.com/30731356/116305865-35b64e00-a759-11eb-8ca0-08af0edebcaf.png)
5. Go back to Tools - Marketing - Sharing Buttons, and enable sharing buttons on Pages. 
6. This will allow the icons to appear on those posts.

#### What I expected to happen
The Sharing Button icons on the Blog Post Block to appear in full, instead of half appearing (appearing, but with the icons broken).

#### What actually happened
The Sharing Buttons appear with broken icons, unless both Posts and Pages are enabled in sharing options.

### Context
User report.
#### Browser / OS version
All.
#### Is this specific to the applied theme? Which one?
Tested and confirmed in multiple themes. Haven't found one that works. 
#### Does this happen on simple or atomic sites or both?
Atomic. Simple sites do not show sharing buttons at all, regardless.
#### Is there any console output or error text?
No.
#### Level of impact (Does it block purchases? Does it affect more than just one site?)
All sharing buttons on Atomic Sites with Blog Posts Blocks
#### Screenshot / Video: If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/30731356/116306173-9776b800-a759-11eb-9bb2-ee3d5ff21bd9.png)

**Comment by synora** [comment]
I'm able to replicate this thanks to your detailed steps @scosgro. @simison this is somewhat related to https://github.com/Automattic/wp-calypso/issues/50034 with sharing buttons working one way on AT sites but not on simple sites.

**Comment by retnonindya** [comment]
I just want to inform that there's a user report: 4232022-zen

**Comment by zdenys** [comment]
Just noting that it's related to https://github.com/Automattic/wp-calypso/issues/46634 which has more comments. The underlying issue may be the same.

**Comment by kathrynwp** [comment]
- [ ] Ran into this in a Quick Start session with Hever. See #4374569-zen

As a workaround, I provided custom CSS to hide the buttons completely.

```
.wp-block-newspack-blocks-homepage-articles .sharedaddy {
  display: none;
}
```

**Comment by DavidSaladu** [comment]
- [ ] 4707132-zd-woothemes

It happened as well in 4707132-zd-woothemes. I noticed that the icons did appear correctly by opening a single post directly, but broke when opened through the blog post block. However, the icons appeared correctly (with page disabled), when the blog post block was within a post.

From this, I understand that if social buttons are disabled for pages, rather than the icons appearing broken, the icons shouldn't appear at all since the blog post block, even if it displays posts, it's within a page. However, in that case, it might be interesting to share that within the FAQ questions of the [Blog post block support page](https://wordpress.com/support/wordpress-editor/blocks/blog-posts-block/).

**Comment by the-misha** [comment]
Another user report: 34162452-hc

- [ ] User would like follow-up via 4774227-zd-woothemes.

**Comment by ai2n** [comment]
Came across this in 35080932-hc

**Comment by villanovachile** [comment]
- [ ] Another report in 5289633-zen

**Comment by jromales** [comment]
- [ ] Another one in 5316700-zd-woothemes 

It happens when sharing buttons are disabled for Pages. I recommended to use CSS Code if they don't want to enable Pages

**Comment by wojtekn** [comment]
Closing as it was fixed in https://github.com/Automattic/jetpack/pull/24896

**Comment by zdenys** [comment]
I encountered another case in 6319093-zd-woothemes of Atomic site with Quadrat White (FSE) theme. 

![Ff70yv.png](https://github.com/Automattic/wp-calypso/assets/39308239/f428e98b-eb2d-4a73-bf56-82fd5b401291)

Only after enabling Sharing Buttons for Pages they started displaying correctly within Blog Posts Block.

Is there a possibility that the underlying issue hasn't been addressed?

**Comment by simison** [comment]
@zdenys could be a new, different problem. Worth opening a new issue since this is years old.

**Comment by zdenys** [comment]
> @zdenys could be a new, different problem. Worth opening a new issue since this is years old.

I was not able to reproduce it on my site yet in the time I had. I'll leave it until there are new reports of this.
